File: minmax.c
#include <stdio.h>
 int min;
 int max;

 void min_max(int arr[], int i, int j){
    int min1, max1;
    if(i == j){
        max = min = arr[i];
    }

   else if (i == j-1){
        if (arr[i] < arr[j]){
            max = arr[j];
            min = arr[i];
        }
        else{
            max = arr[i];
            min = arr[j];
        }
    }

    else{
        int mid;
        mid = (i+j)/2;
        min_max(arr, i, mid);
        max1 = max;
        min1 = min;
        min_max(arr, mid+1, j);
        if (max < max1)
            max = max1;
        if (min > min1)
            min = min1;


    }
 }


 int main(){
    int i, j;
    int arr[] = {2, 23, 45, 56,7383};
    j = sizeof(arr)/sizeof(int);
    min_max(arr, 0, j-1);
    printf("min = %d, max = %d", min, max);
 }

File: python/minmax.py
def minmax(myList,i,j):
    min1=0;
    max1=0;
    if(i==j):
        global min0,max0
        min0=max0=myList[i]
    elif(i==j-1):
        if(myList[i]<myList[j]):
            max0=myList[j]
            min0=myList[i]
        else:
            max0=myList[i]
            min0=myList[j]
    else:
        mid=(i+j)//2
        minmax(myList,i,mid)
        min1=min0
        max1=max0
        minmax(myList,mid+1,j)
        if max0<max1:
            max0=max1
        if min0>min1:
            min0=min1
lists=[2,8,43,1,92,4,7]
length=len(lists)
minmax(lists,0,length-1)
print(f"min={min0} max={max0}")